File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxpro = 0;
        int minsofar = prices[0];
        for (int ele : prices){
            minsofar = Math.min(minsofar, ele);
            int profit = ele - minsofar ;
            maxpro = Math.max(profit, maxpro);
        }
        return maxpro;

    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxpro = 0
        minsofar = prices[0]
        for i in range(len(prices)):
            minsofar = min(minsofar, prices[i])
            profit = prices[i] - minsofar
            maxpro = max(maxpro, profit)
        return maxpro
        